js-classes/inheritance-es5.js

```diff
@@ -0,0 +1,119 @@
+function Car(name, model, year, color, maxSpeed, fuelCapacity = 60, fuelConsumption = 10) {
+    this.name = name;
+    this.model = model;
+    this.year = year;
+    this.color = color;
+    this.maxSpeed = maxSpeed;
+    this.fuelCapacity = fuelCapacity;
+    this.fuelConsumption = fuelConsumption;
+}
+
+Car.prototype.getFullName = function(name, model) {
+    alert("Your car is " + this.name + ': ' + this.model);
+}
+
+Car.prototype.getAge = function(year) {
+    let age = (2019 - this.year);
+    alert('Your car is : ' + age + ' years old');
+}
+
+Car.prototype.changeColor = function(newColor) {
+    if (newColor === this.color) {
+        alert('WTF Man, your car is already ' + newColor);
+    }
+    else {
+        alert('Now your car has ' + newColor + ' color');
+        return newColor;
+    }
+}
+
+Car.prototype.calculateWay = function(kilometers, fuel) {
+    if (fuel < 10) {
+        alert('You need to find gasoline!')
+    }
+    let time = Math.round(kilometers / this.maxSpeed);
+    alert('Your way will take ' + time + 'hours');
+    let requestedFuel = this.fuelConsumption * time;
+    if (requestedFuel > fuel) {
+        let refuelTimes = Math.floor((requestedFuel - fuel) / this.fuelCapacity) + 1;
+        alert('You need to visit gas station ' + refuelTimes + ' times in your trip');
+    }
+}
+
+function BMW(model, version, year, color, wheelDisc, maxSpeed) {
+    this.model = model;
+    this.version = version;
+    this.year = year;
+    this.color = color;
+    this.wheelDisc = wheelDisc;
+    this.maxSpeed = maxSpeed;
+}
+
+BMW.prototype.__proto__ = Car.prototype;
```

Comment:
We **NEVER** use property `__proto__` in js. It is really really bad practice

```suggestion
BMW.prototype = Object.create(Car.prototype);
```